### ISSUE TITLE
[Bug] Need spacing between recipe items

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#3720]
### Description

In both the recipes search results list and My Recipes list, all the recipe items are touching. This is not very accessible because the blending of the images/items makes it difficult to distinguish one recipe item from another. Therefore, spacing should be added between the items, as soon as the controls make it feasible.

### Steps to Reproduce

1. Run Recipes app in Android or iOS
2. Search a recipe or click on any button to search (i.e. "Lunch")
3. Observe the adjacent images in the search results touch / blend together

Similarly:
1. Run Recipes app in Android or iOS
2. Navigate to My Recipes tab
3. Observe the adjacent images in the search results touch / blend together

### Version with bug

Preview 11

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, I was *not* able test on other platforms

### Affected platform versions

all
